exchange/bitswap/bitswap.go

```diff
@@ -208,6 +208,12 @@ func (bs *Bitswap) GetBlocks(ctx context.Context, keys []key.Key) (<-chan *block
 		log.Event(ctx, "Bitswap.GetBlockRequest.Start", &k)
 	}
 
+	go func() {
```

Comment:
+1 to having callers always give the proper timeouts. we really should remove remaining `context.TODO` and `context.Background` to make sure everything has a proper reason for existing.

i believe the http server stuff has ways of detecting that the client has gone away and is no longer expecting a response. it may be tricky to get the wiring to the context right, if it's not yet there.

we may want to hunt into some google http server code that uses contexts, or look in the golang-nuts mailing list.